internal/resource/config/incomingproxysettings/incoming_proxy_settings_resource.go

```diff
@@ -0,0 +1,285 @@
+	ClientCertChainSSLHeaderName  types.String `tfsdk:"client_cert_chain_ssl_header_name"`
+	ProxyTerminatesHttpsConns     types.Bool   `tfsdk:"proxy_terminates_https_conns"`
+}
+
+// GetSchema defines the schema for the resource.
+func (r *incomingProxySettingsResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	schema := schema.Schema{
+		Description: "Manages incoming proxy settings",
+		Attributes: map[string]schema.Attribute{
+			"forwarded_ip_address_header_name": schema.StringAttribute{
+				Description: "Globally specify the header name (for example, X-Forwarded-For) where PingFederate should attempt to retrieve the client IP address in all HTTP requests.",
+				Computed:    false,
+				Optional:    true,
+			},
+			"forwarded_ip_address_header_index": schema.StringAttribute{
+				Description: "PingFederate combines multiple comma-separated header values into the same order that they are received. Define which IP address you want to use. Default is to use the last address.",
+				Computed:    false,
+				Optional:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf("FIRST", "LAST"),
+				},
+			},
+			"forwarded_host_header_name": schema.StringAttribute{
+				Description: "Globally specify the header name (for example, X-Forwarded-Host) where PingFederate should attempt to retrieve the hostname and port in all HTTP requests.",
+				Computed:    false,
+				Optional:    true,
+			},
+			"forwarded_host_header_index": schema.StringAttribute{
+				Description: "PingFederate combines multiple comma-separated header values into the same order that they are received. Define which hostname you want to use. Default is to use the last hostname.",
+				Computed:    false,
+				Optional:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf("FIRST", "LAST"),
+				},
+			},
+			"client_cert_ssl_header_name": schema.StringAttribute{
+				Description: "While the proxy server is configured to pass client certificates as HTTP request headers, specify the header name here.",
+				Computed:    false,
+				Optional:    true,
+			},
+			"client_cert_chain_ssl_header_name": schema.StringAttribute{
+				Description: "While the proxy server is configured to pass client certificates as HTTP request headers, specify the chain header name here.",
+				Computed:    false,
+				Optional:    true,
+			},
+			"proxy_terminates_https_conns": schema.BoolAttribute{
+				Description: "Allows you to globally specify that connections to the reverse proxy are made over HTTPS even when HTTP is used between the reverse proxy and PingFederate.",
+				Computed:    true,
+				Optional:    true,
+			},
+		},
+	}
+
+	id.ToSchema(&schema)
+	resp.Schema = schema
+}
+
+func addOptionalIncomingProxySettingsFields(ctx context.Context, addRequest *client.IncomingProxySettings, plan incomingProxySettingsResourceModel) error {
+
+	if internaltypes.IsDefined(plan.ForwardedIpAddressHeaderName) {
```

Comment:
Since these are all primitive values, you don't need any of the IsDefined checks in this method.

You can also update this method to not return error, since it never returns anything but nil.